interface/login/login.php

```diff
@@ -150,6 +150,14 @@ function imsubmitted() {
                                 </tr>
                             <?php endif; ?>
 
+                            <?php if (isset($_GET['loginfirst'])): ?>
+                                <tr>
+                                    <td colspan='2' class='text' style='color:red'>
+                                        <?php echo xlt('A login is required to access this platform.'); ?>
```

Comment:
This is fine (testing by setting session timeout to 5 minutes), but I would prefer some sort of explicit statement like "You have been logged out due to inactivity, or you have experienced an error.  Please re-enter access credentials."
The "Site ID is missing from session data!" message used to appear in a single frame when the application was open but attempted an action in that frame (frames are now iframe tabs).  This message was not explicit enough, and created confusion.  Booting to the gateway page is much better in some ways, *but if someone was working in a text field in another frame and got booted, they will lose their unsaved data without being able to copy it to the clipboard.*  Having a modal window re-login of some kind would be very helpful in this case, especially when it is an actual server error like mysql going poof.

My arguments above may not be totally coherent, but if we can moderate 1. confusion as to why they were booted out of the system (inactivity), 2. Loss of unsaved data, then that would be the preferred approach.